### PR TITLE
Fix datetimeindex in set_index

### PR DESCRIPTION
**Pull request by phofl**
ref #525

sits on https://github.com/dask/dask/pull/10757

### CODE REVIEW

**Comment by phofl** [comment]
this one should be ready now after the dask/dask pr is in

File: dask_expr/_accessor.py [hunk 1]
```diff
@@ -94,4 +94,5 @@ def operation(obj, accessor, attr, args, kwargs):
 
 class FunctionMapIndex(FunctionMap):
     def _divisions(self):
+        # TODO: We can do better here
```
Comment (milesgranger) [inline dask_expr/_accessor.py:97]: I've seen these todo's, kinda assuming you have a method for dealing with them or do we want to be strict about making follow up issues?

**Review by milesgranger — APPROVED** [review]
Just some more questions, still looks good to me though!

File: dask_expr/_accessor.py (hunk 1 above)
Comment (phofl) [reply dask_expr/_accessor.py:97]: grepping when we are happy with the state of the test suite,

this specific occurrence isn't super important, so I avoided creating an issue, it's a tiny annoyance for now

**Comment by phofl** [comment]
I did not know that we were caching environments....

**Pull request merged**